config/v2/resolvers.go

```diff
@@ -0,0 +1,188 @@
+package v2
+
+import (
+	"github.com/chanzuckerberg/fogg/config/v1"
+)
+
+// lastNonNil, despite its name can return nil if all results are nil
+func lastNonNil(getter func(Common) *string, commons ...Common) *string {
+	var s *string
+	for _, c := range commons {
+		t := getter(c)
+		if t != nil {
+			s = t
+		}
+	}
+	return s
+}
+
+// lastNonNilInt, despite its name can return nil if all results are nil
+func lastNonNilInt64(getter func(Common) *int64, commons ...Common) *int64 {
+	var s *int64
+	for _, c := range commons {
+		t := getter(c)
+		if t != nil {
+			s = t
+		}
+	}
+	return s
+}
+
+// lastNonNilStringSlice, despite its name can return nil if all results are nil
+func lastNonNilStringSlice(getter func(Common) []string, commons ...Common) []string {
+	var s []string
+	for _, c := range commons {
+		t := getter(c)
+		if t != nil {
+			s = t
+		}
+	}
+	return s
+}
+
+// ResolveRequiredString will resolve the value and panic if it is nil. Only to be used after validations are run.
+func ResolveRequiredString(getter func(Common) *string, commons ...Common) string {
+	return *lastNonNil(getter, commons...)
+}
+
+// ResolveRequiredInt will resolve the value and panic if it is nil. Only to be used after validations are run.
+func ResolveRequiredInt64(getter func(Common) *int64, commons ...Common) int64 {
+	return *lastNonNilInt64(getter, commons...)
+}
+
+func ResolveOptionalString(getter func(Common) *string, commons ...Common) *string {
+	return lastNonNil(getter, commons...)
+}
+
+func ResolveOptionalStringSlice(getter func(Common) []string, commons ...Common) []string {
+	return lastNonNilStringSlice(getter, commons...)
+}
+
+func ResolveStringArray(def []string, override []string) []string {
+	if override != nil {
+		return override
+	}
+	return def
+}
+
+func ResolveStringMap(getter func(Common) map[string]string, commons ...Common) map[string]string {
+	resolved := map[string]string{}
+
+	for _, c := range commons {
+		m := getter(c)
+		for k, v := range m {
+			resolved[k] = v
+		}
+	}
+	return resolved
+}
+
+// ResolveAWSProvider will return an AWSProvder iff one of the required fields is set somewhere in the set of Common
+// config objects passed in. Otherwise it will return nil.
+func ResolveAWSProvider(commons ...Common) *AWSProvider {
+
+	profile := lastNonNil(AWSProviderProfileGetter, commons...)
+	region := lastNonNil(AWSProviderRegionGetter, commons...)
+	version := lastNonNil(AWSProviderVersionGetter, commons...)
+
+	if profile != nil || region != nil || version != nil {
+		return &AWSProvider{
+			Profile: profile,
+			Region:  region,
+			Version: version,
+
+			// optional fields
+			AccountID:         lastNonNilInt64(AWSProviderAccountIdGetter, commons...),
+			AdditionalRegions: ResolveOptionalStringSlice(AWSProviderAdditionalRegionsGetter, commons...),
+		}
+	}
+	return nil
+}
+
+func OwnerGetter(comm Common) *string {
+	return comm.Owner
```

Comment:
Maybe this can be comm.GetOwner()?

Reply:
Same with the others

Reply:
Yeah though we use `func (comm Common) *string` in a bunch of places.